src/peer_connection.c

```diff
@@ -268,8 +268,6 @@ static void peer_connection_state_new(PeerConnection* pc, DtlsSrtpRole role) {
 
   pc->sctp.connected = 0;
 
-  agent_clear_candidates(&pc->agent);
```

Comment:
Do we need this @sepfy? To allow users to re-use a PeerConnection? What is the flow for that.

I can add tests and make sure behavior doesn't break.

Reply:
Yes. My use case is that when the peer connection is in certain states (e.g., connectivity check failure or after disconnection), the signaling service can initiate the process to create an offer again, re-establishing the connection.
[peer_connection_create_offer](https://github.com/sepfy/libpeer/blob/main/src/peer_signaling.c#L126)

Reply:
@sepfy Can we have it so `peer_connection_state_new` only clears the candidate if a new offer is being created?

If `peer_connection_set_remote_description` is called it will not clear them?

Reply:
ok! Have you verified if this modification works for you? if it's ok, i'll merge it.